lib/morpheus.gi

```diff
@@ -367,300 +367,655 @@ end);
 ##
 # try to find a small faithful action for an automorphism group
 InstallGlobalFunction(AssignNiceMonomorphismAutomorphismGroup,function(au,g)
-local hom, allinner, gens, c, ran, r, cen, img, dom, u, subs, orbs, cnt, br, bv,
-v, val, o, i, comb, best,actbase,action;
+local hom, allinner, gens, c, ran, r, cen, img, dom, u, subs, orbs, cnt,
+      doelms,ser,pos, v, val, o, i, j, comb, best,actbase,action,finish,
+      bestdeg,deg,baddegree,auo,of,cl,store, offset,claselms,fix,new,
+      preproc,postproc;
+
+  finish:=function(hom)
+    SetIsGroupHomomorphism(hom,true);
+    SetIsBijective(hom,true);
+    SetFilterObj(hom,IsNiceMonomorphism);
+    SetNiceMonomorphism(au,hom);
+    SetIsHandledByNiceMonomorphism(au,true);
+  end;
+
+  # avoid storing list of class elements anew
+  claselms:=function(clas)
+    if HasAsSSortedList(clas) then
+      return AsSSortedList(clas);
+    else
+      return Immutable(Set(Orbit(g,Representative(clas))));
```

Comment:
```suggestion
      return MakeImmutable(Set(Orbit(g,Representative(clas))));
```
This avoids making a copy of the set, instead the set is marked as immutable.

Reply:
Thanks. Will change